Xamarin.Forms.Platform.iOS/Renderers/LabelRenderer.cs

```diff
@@ -156,6 +163,53 @@ protected override void OnElementPropertyChanged(object sender, PropertyChangedE
 				UpdateAlignment();
 		}
 
+		void UpdateTextDecorations()
+		{
+			var textDecorations = Element.TextDecorations;
+			if (Element.IsSet(Label.TextDecorationsProperty))
+			{
+#if __MOBILE__
+				var newAttributedText = new NSMutableAttributedString(Control.AttributedText);
+				var strikeThroughStyleKey = UIStringAttributeKey.StrikethroughStyle;
+				var underlineStyleKey = UIStringAttributeKey.UnderlineStyle;
+
+#else
+				var newAttributedText = new NSMutableAttributedString(Control.AttributedStringValue);
+				var strikeThroughStyleKey = NSStringAttributeKey.StrikethroughStyle;
+				var underlineStyleKey = NSStringAttributeKey.UnderlineStyle;
+#endif
+				var range = new NSRange(0, newAttributedText.Length);
+
+				if (_strikeSet)
```

Comment:
Just because I haven't delved into this before, do you have to wipe everything that add it again, every time?

Reply:
On some platforms there wasn't an easy way to check if a particular style was set, so you would have to remove them all and then add back the ones you want. I would think that 99% of the time we wouldn't be combining Underline and strike together so this shouldn't cause unnecessary overhead of removing stuff that we don't need to.

Technically for the enum-based ones I could take the approach of only removing them if they are not in the BP, but then I would also presumbaly have to add logic to not add them in this case. That seemed a little over the top to me, but I can switch those if you want. For iOS and WPF I don't think there's really a good way to not do this though.